# Laboratorio Pandas (EDA y Unión de Datos)

## Los datos

Para este laboratorio, trabajarás con los siguientes conjuntos de datos:

- **`netflix_originals.csv`**: Contiene información sobre producciones originales de Netflix. Las columnas que nos encontraremos en este conjunto de datos son: 

   - `Title`: Contiene el nombre del título de la producción original de Netflix.

   - `Genre`: Especifica el género o combinación de géneros de la producción.

   - `Premiere`: Indica la fecha de estreno de la producción en Netflix.

   - `Runtime`: Contiene la duración de la producción en minutos.

   - `IMDB Score`: Representa la calificación promedio que la producción ha recibido en la plataforma IMDB.

   - `Language`: Especifica el idioma principal o los idiomas en los que se presenta la producción.

- **`netflix_titles.csv`**: Contiene información sobre una variedad de títulos disponibles en la plataforma Netflix, incluyendo tanto películas como series de televisión. Las columnas que nos encontraremos en este conjunto de datos son:

   - `show_id`: Identificador único para cada título en el dataset.

   - `type`: Especifica el tipo de contenido, ya sea "Movie" (película) o "TV Show" (serie de televisión).

   - `title`: Contiene el nombre del título de la producción.

   - `director`: Muestra el nombre del director del título. Puede estar vacío en algunos casos.

   - `cast`: Lista de actores y actrices que participaron en el título. También puede estar vacío en algunos casos.

   - `country`: País o países donde se produjo el título. Esta columna también puede tener valores faltantes.

   - `date_added`: Fecha en la que el título fue añadido al catálogo de Netflix.

   - `release_year`: Año en que el título fue lanzado o producido.

   - `rating`: Clasificación del contenido en términos de audiencia (por ejemplo, PG-13, TV-MA).

   - `duration`: Duración del título. Para películas, se expresa en minutos (e.g., "90 min") y para series en temporadas (e.g., "2 Seasons").

   - `listed_in`: Categorías o géneros bajo los cuales el título está clasificado (por ejemplo, "Documentaries", "TV Dramas").

   - `description`: Breve sinopsis o descripción del contenido del título.

A lo largo de los laboratorios de Pandas, trabajarás con estos conjuntos de datos (o con los derivados del Laboratorio), algunas de las actividades que realizaremos son:

- Exploración de datos y análisis exploratorio (EDA) para familiarizarse con los conjuntos de datos.

- Realización de uniones entre conjuntos de datos, aplicando diferentes técnicas de *merging* y analizando la relevancia de cada método.

- Uso de funciones de selección y filtrado (`loc`, `iloc`) para extraer información clave.

- Desarrollo de análisis descriptivos y visualización de patrones para comprender mejor las decisiones estratégicas de la plataforma.

----

## Objetivo General de la Serie de Laboratorios



1. **Unión de `netflix_titles.csv` con `netflix_originals.csv`:**

   - **Análisis de contenido original:**  Identificarás los títulos que son producciones originales de Netflix y los compararás con los títulos adquiridos. Este análisis te permitirá:

     - Comparar la diversidad de géneros entre los títulos originales y no originales.

     - Evaluar si ciertos géneros o tipos de contenido (como documentales o series) son más frecuentes en las producciones originales.

     - Investigar si la producción original se concentra en determinados países o idiomas.


   - **Evaluación de contenido:** En esta etapa, explorarás si las producciones originales de Netflix tienden a recibir mejores o peores evaluaciones en comparación con el contenido adquirido. Algunas preguntas que se buscarán responder incluyen:

     - ¿Los títulos originales tienen una calificación promedio superior en plataformas como IMDB?

     - ¿Son las producciones originales más consistentes en términos de calidad percibida?

     - ¿Existen diferencias en la popularidad de los títulos originales según su género o país de origen?


   - **Identificación de tendencias:**  Por último, este análisis te permitirá explorar cómo ha evolucionado la estrategia de contenido de Netflix a lo largo del tiempo. Algunos puntos clave a investigar serán:

     - ¿Cómo ha crecido la producción original en comparación con la adquisición de contenido?

     - ¿Qué géneros o tipos de contenido han recibido mayor inversión en los últimos años?
     
     - ¿Existen patrones en los países o idiomas de las producciones originales, especialmente en relación con la expansión global de Netflix?


----


## Ejercicio: Análisis y Enriquecimiento del Catálogo de Netflix

### Parte 1: Análisis Exploratorio de Datos (EDA)

1. **Carga de los datos:**

   - Deberás cargar los dos conjuntos de datos que te hemos proporcionado para realizar este Laboratorio:

     - `netflix_originals.csv`

     - `netflix_titles.csv`

   Deberás revisar la estructura de cada conjunto de datos utilizando los métodos aprendidos durante la lección.

2. **Análisis de columnas y consistencia:**

   - Deberás hacer un análisis exploratorio de cada uno de los conjuntos de datos. Será obligatorio que incluyas las conclusiones generales obtenidas de cada uno de los análisis exploratorios en el Jupyter Notebook de tus soluciones.

   - Explorar qué columnas contiene cada conjunto de datos y cuáles son sus tipos de datos. Identificar si hay valores nulos o duplicados.

   - Algunas preguntas que nos podemos plantear en esta fase son (recuerda que esto es solo un ejemplo, y que podemos explorar todo lo que queramos/necesitemos para entender lo mejor posible los conjuntos de datos):

      - ¿Cuáles son los tipos de datos de cada columna? Nos puede ayudar a detectar si algún tipo de dato está incorrectamente asignado, como fechas almacenadas como cadenas de texto.

      - ¿Existen valores nulos? ¿En qué columnas están? Este paso es crucial para decidir si deben ser eliminados, rellenados o ignorados.

      - ¿Cuántos títulos únicos hay en cada conjunto de datos? Nos puede permitir verificar si existen títulos duplicados que puedan distorsionar el análisis.

      - ¿Qué géneros tenemos en el conjunto de datos? Esto nos puede ayudar a identificar los géneros más populares o incluso a identificar una posible limpieza de dicha columna para poder extraer conclusiones más claras en nuestro análisis.

### Parte 2: Unión de Datos

1. **Realización de la unión:**

   - Deberás realizar la unión que consideres más apropiada entre `netflix_titles.csv` y `netflix_originals.csv`. Además, deberás justificar el tipo de unión que has realizado.

   - Esto nos ayudará a contestar preguntas más adelente del tipo:

     - ¿Cuántos títulos en `netflix_titles.csv` son producciones originales de Netflix?

     - ¿Hay diferencias significativas en la puntuación de IMDB entre títulos originales y no originales?

### Parte 3: Uso de `loc` e `iloc`

Usando el DataFrame creado en la Parte 2: 

1. **Filtrado de datos con `loc`:**

   - Selecciona todas las filas donde el tipo de contenido sea “Movie”.

   - Muestra solo los títulos (title) y la duración (duration) de todos los contenidos que sean de tipo “TV Show”.

   - Selecciona todas las filas donde el país sea “United States”.

   - Muestra todas las películas (type = “Movie”) y selecciona las columnas title y director.

   - Selecciona los títulos (title) y géneros (listed_in) de todas las producciones lanzadas en 2018.

   - Selecciona las filas donde la columna director esté vacía y muestra solo los títulos (title).

2. **Acceso a datos específicos con `iloc`:**

   - Selecciona las primeras 5 filas del conjunto de datos y muestra solo las columnas title, director y country.

   - Muestra las últimas 5 filas del conjunto de datos y selecciona las columnas title y description.

   - Selecciona las primeras 10 filas del dataset y muestra las columnas desde la segunda hasta la quinta (incluidas). 

   - Muestra las últimas 7 filas del dataset y selecciona las columnas title, director, y country usando índices posicionales.

   - Selecciona las filas desde la 15 hasta la 25 (incluidas) y muestra las primeras 4 columnas.

   - Muestra las filas 20 a 30 y solo las columnas title, release_year, y rating usando índices posicionales.


**NOTA IMPORTANTE**: Antes de terminar el lab, guarda el conjunto de datos creado en la fase de unión para trabajar con el en las siguientes lecciones. 

<div style="text-align: center;">
  <img src="https://github.com/Hack-io-Data/Imagenes/blob/main/01-LogosHackio/logo_naranja@4x.png?raw=true" alt="esquema" />
</div>

In [119]:
import pandas as pd

In [120]:
pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", None)

In [121]:
df_originals = pd.read_csv("../DATOS/netflix_originals.csv", index_col=0)
df_titles = pd.read_csv("../DATOS/netflix_titles.csv", index_col=0)

In [171]:
df_originals.head(1)


Title        Genre        Premiere  Runtime  IMDB Score  \
0  Enter the Anime  Documentary  August 5, 2019       58         2.5   

           Language Originals  
0  English/Japanese        Sí

In [172]:
df_titles.head(1)

show_id   type                 title         director cast        country  \
0      s1  Movie  Dick Johnson Is Dead  Kirsten Johnson  NaN  United States   

           date_added  release_year rating duration      listed_in  \
0  September 25, 2021          2020  PG-13   90 min  Documentaries   

                                         description  
0  As her father nears the end of his life, filmm...

## NETFLIX_ORIGINALS

Cargamos el archivo netflix_originalas para empezar con el análisis exploratorio.

In [122]:
df_originals.head()

Title                  Genre           Premiere  Runtime  \
0  Enter the Anime            Documentary     August 5, 2019       58   
1      Dark Forces               Thriller    August 21, 2020       81   
2          The App  Science fiction/Drama  December 26, 2019       79   
3   The Open House        Horror thriller   January 19, 2018       94   
4      Kaali Khuhi                Mystery   October 30, 2020       90   

   IMDB Score          Language  
0         2.5  English/Japanese  
1         2.6           Spanish  
2         2.6           Italian  
3         3.2           English  
4         3.4             Hindi

Para asegurarnos que, cuando juntemos las dos bases de datos, podamos diferenciar las películas originales, añadimos una columna llamada ''Originals'' y que responda Sí en todas las filas.

In [123]:
df_originals["Originals"] = "Sí"

In [124]:
df_originals.head()

Title                  Genre           Premiere  Runtime  \
0  Enter the Anime            Documentary     August 5, 2019       58   
1      Dark Forces               Thriller    August 21, 2020       81   
2          The App  Science fiction/Drama  December 26, 2019       79   
3   The Open House        Horror thriller   January 19, 2018       94   
4      Kaali Khuhi                Mystery   October 30, 2020       90   

   IMDB Score          Language Originals  
0         2.5  English/Japanese        Sí  
1         2.6           Spanish        Sí  
2         2.6           Italian        Sí  
3         3.2           English        Sí  
4         3.4             Hindi        Sí

Identificación de variables: identificamos todas las variables a través de las columnas presentes en la tabla. Así podremos comprender qué aspectos se registran o se miden.

In [125]:
df_originals.columns

Index(['Title', 'Genre', 'Premiere', 'Runtime', 'IMDB Score', 'Language',
       'Originals'],
      dtype='object')

Tipos de datos: para saber qué tipos de datos tiene cada variable, lo desglosamos a través de .info() para poder estudiarlos.

In [126]:
df_originals.info()

<class 'pandas.core.frame.DataFrame'>
Index: 513 entries, 0 to 583
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Title       513 non-null    object 
 1   Genre       513 non-null    object 
 2   Premiere    513 non-null    object 
 3   Runtime     513 non-null    int64  
 4   IMDB Score  513 non-null    float64
 5   Language    513 non-null    object 
 6   Originals   513 non-null    object 
dtypes: float64(1), int64(1), object(5)
memory usage: 32.1+ KB


#### Datos estadísticos

In [127]:
round(df_originals.describe(),2)

Runtime  IMDB Score
count   513.00      513.00
mean     94.67        6.21
std      26.65        0.97
min       4.00        2.50
25%      87.00        5.60
50%      97.00        6.30
75%     108.00        6.90
max     209.00        9.00

En los datos estadísticos de las variables numéricas podemos observar lo siguiente: 

**count**: Vemos que en ninguna de las dos columnas hay valores nulos.
mean: Vemos la media de la duración de la producción en minutos (runtime) y de la puntuación en IMDB (IMDB score).
**std**: La desviación estándar de ambos datos. Aquí podemos observar lo siguiente: 

- `RUNTIME`: La desviación estándar es 26.65, lo que indica una dispersión considerable respecto a la media.
Esto significa que los datos están relativamente dispersos: muchos valores probablemente están a más de 26.6 unidades del promedio. Es posible que haya mayor heterogeneidad en los datos.

- `IMDB score`: La desviación estándar es 0.97, lo que indica poca dispersión alrededor de la media.
Los datos están más concentrados cerca del promedio, mostrando menor variabilidad.

**min**: El valor mínimo de cada una de las variables. Podemos ver que la puntuación más baja de IMDB es de 2.50.

**25%**: Representa el valor por debajo del cual se encuentra el 25% del tiempo de producción y que solo el 25% de las puntuaciones están por debajo de 5.60.

**50%**: Representa la mediana de ambas variables.

**75%**: El valor por debajo del cual están el 75% de los datos. Podemos decir que la mayoría de las producciones están por debajo de 6.90 de puntuación en IMDB.

**max**: El valor máximo de cada variable en el conjunto de datos.

#### Datos categóricos

In [128]:
df_originals.describe(include="object")

Title        Genre          Premiere Language Originals
count               513          513               513      513       513
unique              513          106               350       37         1
top     Enter the Anime  Documentary  November 1, 2019  English        Sí
freq                  1          132                 5      352       513

En el caso de los valores categóricos podemos sacar las siguientes conclusiones: 

**count**: Numero de observaciones no nulas en cada columna. No hay ninguna columna con nulos.

**unique**: Podemos apreciar como hay un total de 106 géneros diferentes y 37 idiomas.

**top**: El valor más común de cada columna. "Enter de Anime" es el título más repetido, "Documentary" el género, "November 1, 2019" la fecha y "English" el idioma.

**freq**: La frecuencia del valor más común. Destacamos el ingés que está en un total de 352 valores.

Para analizar los valores únicos más llamativos, nos vamos a centrar en la columna Genre que es la más importante en este aspecto:

In [129]:
pd.DataFrame(df_originals["Genre"].value_counts().head(10))

count
Genre                 
Documentary        132
Drama               73
Comedy              42
Romantic comedy     35
Thriller            33
Comedy-drama        14
Crime drama         10
Horror               9
Biopic               8
Action               7

Aquí podemos ver el top 10 de géneros más repetidos y podemos afirmar que los "Documentary" son los más repetidos y con bastante recurrencia, representan casi el doble con respecto al los "Drama" que están en segundo lugar. Por último, cierra el top 3 los "Comedy".

In [130]:
df_cat = df_originals.select_dtypes(include = "object")
df_cat.head()

Title                  Genre           Premiere  \
0  Enter the Anime            Documentary     August 5, 2019   
1      Dark Forces               Thriller    August 21, 2020   
2          The App  Science fiction/Drama  December 26, 2019   
3   The Open House        Horror thriller   January 19, 2018   
4      Kaali Khuhi                Mystery   October 30, 2020   

           Language Originals  
0  English/Japanese        Sí  
1           Spanish        Sí  
2           Italian        Sí  
3           English        Sí  
4             Hindi        Sí

Hacemos un bucle for para iterar entre todos los datos de las columnas categóricas. En este caso, al haber muchos datos es un poco difícil de leer, pero podemos estudiar cada dato de una manera muy extensa.

In [160]:
columas_categoricas = df_cat.columns

for columna in columas_categoricas:
    print(f"ANÁLISIS DE LA COLUMNA {columna.upper()}:")
    print(f"Las frecuencias de los valores únicos de las categorías son: {df_cat[columna].value_counts().head(10)}")
    print("      ")

ANÁLISIS DE LA COLUMNA TITLE:
Las frecuencias de los valores únicos de las categorías son: Title
Enter the Anime                       1
Sand Castle                           1
Paper Lives                           1
Night in Paradise                     1
Moxie                                 1
Malcolm & Marie                       1
Little Miss Sumo                      1
Just Another Christmas                1
Imperial Dreams                       1
Hope Frozen: A Quest to Live Twice    1
Name: count, dtype: int64
      
ANÁLISIS DE LA COLUMNA GENRE:
Las frecuencias de los valores únicos de las categorías son: Genre
Documentary        132
Drama               73
Comedy              42
Romantic comedy     35
Thriller            33
Comedy-drama        14
Crime drama         10
Horror               9
Biopic               8
Action               7
Name: count, dtype: int64
      
ANÁLISIS DE LA COLUMNA PREMIERE:
Las frecuencias de los valores únicos de las categorías son: Premiere
Novembe

También utilizaremos el método .dtypes() para saber qué tipos de datos tiene cada columna.

In [132]:
pd.DataFrame(df_originals.dtypes)

0
Title        object
Genre        object
Premiere     object
Runtime       int64
IMDB Score  float64
Language     object
Originals    object

Podemos ver que nuestro DataFrame tiene todos los valores de tipo objetc (strings), excepto la columna Runtime que, como sabemos, corresponde a la duración de la producción en minutos.

En este caso no vamos a analizar los nulos, porque como vimos en el análsis no hay ninguna variable con nulos.

In [133]:
nulos = pd.DataFrame(df_originals.isnull().sum())
nulos.columns = ["Total nulos"]
nulos

Total nulos
Title                 0
Genre                 0
Premiere              0
Runtime               0
IMDB Score            0
Language              0
Originals             0

Por último estudiaremos si hay valores duplicados en nuestro DataFrame.

In [134]:
df_originals.duplicated().sum()

np.int64(0)

## NETFLIX_TITLES


In [135]:
df_titles.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA        NaN   
3  September 24, 2021          2021  TV-MA        NaN   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

Identificación de variables: identificamos todas las variables a través de las columnas presentes en la tabla. Así podremos comprender qué aspectos se registran o se miden.

In [136]:
df_titles.columns

Index(['show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in', 'description'],
      dtype='object')

Tipos de datos: para saber qué tipos de datos tiene cada variable, lo desglosamos a través de .info() para poder estudiarlos.

In [137]:
df_titles.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      3994 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 894.5+ KB


#### Datos estadísticos

In [138]:
round(df_titles.describe())

release_year
count        8807.0
mean         2014.0
std             9.0
min          1925.0
25%          2013.0
50%          2017.0
75%          2019.0
max          2021.0

En este caso, los datos estadísticos no son muy representativos ya que son fechas y no podemos sacar muchas conclusiones al respecto mñas allá de: 

**min**: el año mínimo es el 1925, es decir, la película más antigua es de 1925.

**25%**: Representa el valor por debajo del cuál se encuentra el 25% de las películas que se han hecho antes de 2013.

**50%**: La mediana de la variable, siendo en 2017

**75%**: El valor por debajo del cuál están el 75% de los datos. Es decir, el 75% de las películas son de antes de 2019 y solo el 25% es después de 2019.

**max**: La película más moderna es de 2021.



#### Datos categóricos

In [139]:
pd.DataFrame(df_titles.describe(include="object").T)

count unique                                                top  \
show_id      8807   8807                                                 s1   
type         8807      2                                              Movie   
title        8807   8807                               Dick Johnson Is Dead   
director     6173   4528                                      Rajiv Chilaka   
cast         7982   7692                                 David Attenborough   
country      7976    748                                      United States   
date_added   8797   1767                                    January 1, 2020   
rating       8803     17                                              TV-MA   
duration     3994    185                                          2 Seasons   
listed_in    8807    514                       Dramas, International Movies   
description  8807   8775  Paranormal activity at a lush, abandoned prope...   

             freq  
show_id         1  
type         6131  
title           1  
director       19  
cast           19  
country      2818  
date_added    109  
rating       3207  
duration      425  
listed_in     362  
description     4

En el caso de los valores categóricos podemos sacar las siguientes conclusiones: 

**count**: Numero de observaciones no nulas en cada columna. En este caso si hay alguna columna con nulos (director, cast, country, date_added y duration). Procederemos a analizarlo más adelante.

**unique**: Destacamos las columnas "type" que tiene dos valores únicos, "rating" que tiene 17 valores únicos, "listed_in" que tiene 514 valores únicos divididos en géneros y "duration" con 185 valores únicos.

**top**: El valor más común de cada columna. Combinándolo con la frecuencia podemos descatar las variables "Movie" y "United States"

**freq**: La frecuencia del valor más común.

Para analizar los valores únicos más llamativos, vamos a ir columna por columna analizando los que más nos interesen:

In [141]:
df_titles.head(1)

show_id   type                 title         director cast        country  \
0      s1  Movie  Dick Johnson Is Dead  Kirsten Johnson  NaN  United States   

           date_added  release_year rating duration      listed_in  \
0  September 25, 2021          2020  PG-13   90 min  Documentaries   

                                         description  
0  As her father nears the end of his life, filmm...

In [144]:
df_titles["rating"].unique()

array(['PG-13', 'TV-MA', 'PG', 'TV-14', 'TV-PG', 'TV-Y', 'TV-Y7', 'R',
       'TV-G', 'G', 'NC-17', '74 min', '84 min', '66 min', 'NR', nan,
       'TV-Y7-FV', 'UR'], dtype=object)

In [146]:
pd.DataFrame(df_titles["rating"].value_counts())

count
rating         
TV-MA      3207
TV-14      2160
TV-PG       863
R           799
PG-13       490
TV-Y7       334
TV-Y        307
PG          287
TV-G        220
NR           80
G            41
TV-Y7-FV      6
NC-17         3
UR            3
74 min        1
84 min        1
66 min        1

In [154]:
pd.DataFrame(df_titles["duration"].value_counts().head(10))

count
duration        
2 Seasons    425
90 min       152
98 min       120
104 min      104
107 min       98
110 min       97
116 min       80
112 min       74
85 min        73
113 min       69

In [155]:
pd.DataFrame(df_titles["listed_in"].value_counts().head(10))

count
listed_in                                              
Dramas, International Movies                        362
Documentaries                                       359
Stand-Up Comedy                                     334
Comedies, Dramas, International Movies              274
Dramas, Independent Movies, International Movies    252
Kids' TV                                            220
Children & Family Movies                            215
Children & Family Movies, Comedies                  201
Documentaries, International Movies                 186
Dramas, International Movies, Romantic Movies       180

In [158]:
df_cat1 = df_titles.select_dtypes(include = "object").head()

In [161]:
columas_categoricas1 = df_cat1.columns

for columna in columas_categoricas1:
    print(f"ANÁLISIS DE LA COLUMNA {columna.upper()}:")
    print("      ")
    print(f"Las frecuencias de los valores únicos de las categorías son: {df_cat1[columna].value_counts().head(15)}")
    print("      ")

ANÁLISIS DE LA COLUMNA SHOW_ID:
      
Las frecuencias de los valores únicos de las categorías son: show_id
s1    1
s2    1
s3    1
s4    1
s5    1
Name: count, dtype: int64
      
ANÁLISIS DE LA COLUMNA TYPE:
      
Las frecuencias de los valores únicos de las categorías son: type
TV Show    4
Movie      1
Name: count, dtype: int64
      
ANÁLISIS DE LA COLUMNA TITLE:
      
Las frecuencias de los valores únicos de las categorías son: title
Dick Johnson Is Dead     1
Blood & Water            1
Ganglands                1
Jailbirds New Orleans    1
Kota Factory             1
Name: count, dtype: int64
      
ANÁLISIS DE LA COLUMNA DIRECTOR:
      
Las frecuencias de los valores únicos de las categorías son: director
Kirsten Johnson    1
Julien Leclercq    1
Name: count, dtype: int64
      
ANÁLISIS DE LA COLUMNA CAST:
      
Las frecuencias de los valores únicos de las categorías son: cast
Ama Qamata, Khosi Ngema, Gail Mabalane, Thabang Molaba, Dillon Windvogel, Natasha Thahane, Arno Gre

In [162]:
pd.DataFrame(df_titles.dtypes)

0
show_id       object
type          object
title         object
director      object
cast          object
country       object
date_added    object
release_year   int64
rating        object
duration      object
listed_in     object
description   object

In [166]:
nulos1 = pd.DataFrame(df_titles.isnull().sum())
nulos1.columns = ["Total nulos"]
nulos1

Total nulos
show_id                 0
type                    0
title                   0
director             2634
cast                  825
country               831
date_added             10
release_year            0
rating                  4
duration             4813
listed_in               0
description             0

In [167]:
df_titles.duplicated().sum()

np.int64(0)